lib/evmone/eof.cpp

```diff
@@ -13,6 +13,92 @@ namespace
 {
 constexpr uint8_t FORMAT = 0xef;
 constexpr uint8_t MAGIC = 0x00;
+constexpr uint8_t TERMINATOR = 0x00;
+constexpr uint8_t CODE_SECTION = 0x01;
+constexpr uint8_t DATA_SECTION = 0x02;
+
+std::pair<EOF1Header, EOFValidationErrror> validate_eof_headers(bytes_view code) noexcept
+{
+    enum class State
+    {
+        section_id,
+        section_size,
+        terminated
+    };
+
+    auto state = State::section_id;
+    uint8_t section_id = 0;
+    size_t section_sizes[3] = {0, 0, 0};
+    const auto code_end = code.end();
+    auto it = code.begin() + sizeof(MAGIC) + 2;  // FORMAT + MAGIC + VERSION
+    while (it != code_end && state != State::terminated)
```

Comment:
Why did you decide to implement the generic loop as opposed the short two checks we have in the EIP? Did we ever measure any speed difference?

Reply:
It became handy at one point when implementing RJUMPTABLE with additional jump table section.

That idea is abandoned now, but it can become handy again when we add type section for EOF Functions.

But I don't mind dropping this commit for now.

Reply:
Actually it was written originaly by @chfast in https://github.com/ethereum/evmone/pull/303, I decided not to write another one.